pallets/credit-score/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		CreditScoreReport(T::AccountId, u32),
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn report_credit_score(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			score: u32,
+		) -> DispatchResult {
+			// TODO add the is_registered_enclave as a trait in teerex pallet.
```

Comment:
Using the [is_registered_enclave](https://github.com/integritee-network/pallets/blob/master/teerex/src/lib.rs#L301) without trait is not possible?

Reply:
To use is_registered_enclave, as I know, we can define the pallet inherited from pallet_teerex or define a trait, then binding the trait in the runtime.

Reply:
I don't have much experience with cross-pallet interaction yet, so I can't really help out here.
I only know that in our teeracle pallet we are using the following:
https://github.com/integritee-network/pallets/blob/master/teeracle/src/lib.rs#L161

Reply:
you can call it in teeracle because the code 
pub trait Config: frame_system::Config + pallet_teerex::Config 
but two pallets are tightly coupled, trait is better solution.